app/models/base_facility.rb

```diff
@@ -116,6 +128,12 @@ def per_page
     def max_per_page
       100
     end
+
+    def suggested_names(facility_types, name_part)
+      BaseFacility.where(
+        facility_type: facility_types.map { |t| TYPE_NAME_MAP[t] }
+      ).where('name ILIKE ?', "%#{name_part}%")
```

Comment:
Maybe a conscious business decision, but any consideration for querying on city?